src/mmc.c

```diff
@@ -217,36 +224,134 @@ int mmc_init(unsigned int id)
 	}
 
 	if (!retries) {
-		SERIAL_ERR(ERR_MMC_INIT);
+		SERIAL_ERR(ERR_MMC_SEND_OP_COND);
 		return -1;
 	}
 
+	bool is_sd = 1;
+
+	if (ret) {
+		if (ret == ERR_MMC_TIMEOUT) {
+			SERIAL_PUTS("trying mmc\n");
+
+			uint32_t ocr;
+			uint32_t arg = 0;
+			for (retries = 1000; retries; retries--)
+			{
+				ret = mmc_cmd(id, CMD_SEND_OP_COND, arg, 0x0, MSC_RESPONSE_R3, resp);
+
+				if (ret)
+					break;
+
+				ocr = (resp[0] >> 8) | (resp[1] << 8) | (resp[2] << 24);
+
+				if (ocr & MMC_OCR_BUSY)
+					break;
+
+				arg = ocr | MMC_OCR_HCS;
+
+				udelay(1000);
+			}
+
+			if (!retries || ret) {
+				int err;
+				if (!retries)
+					err = ERR_MMC_SEND_OP_COND_TOUT;
+				else
+					err = ERR_MMC_SEND_OP_COND_ERR;
+				SERIAL_ERR(err);
+				return -1;
+			}
+
+			SERIAL_PUTS_ARGH("OCR = ", ocr, "\n");
```

Comment:
Remove this.

Reply:
I like additional output, but as you wish